sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2AlterTableCommands.scala

```diff
@@ -309,6 +316,27 @@ case class AddConstraint(
   protected def withNewChildInternal(newChild: LogicalPlan): LogicalPlan = copy(table = newChild)
 }
 
+/**
+ * The logical plan of the ALTER TABLE ... ADD CONSTRAINT command for Check constraints.
+ */
+case class AddCheckConstraint(
+    override val table: LogicalPlan,
```

Comment:
this is not really a table, but a validation query plan.

Reply:
We should either not extend `AlterTableCommand` for `AddCheckConstraint`, or have both the `table` and `validationPlan` fields.

Reply:
The current code is fragile as we have code like the following
```
      case a: AlterTableCommand if a.table.resolved && hasUnresolvedFieldName(a) =>
        val table = a.table.asInstanceOf[ResolvedTable]
```

Reply:
Updated. 
The code in `CheckAnalysis` looks a bit ugly now. 
Do we consider remove the `final` from 
```
trait AlterTableCommand extends UnaryCommand {
  def changes: Seq[TableChange]
  def table: LogicalPlan
  final override def child: LogicalPlan = table
}
```